### PR TITLE
iox-#1295 Rework `icediscovery` example

### LINKED ISSUE #1295: Release validation for `v3.0`
## Website/Markdown settings

* [x]  ~~Add `.markdownlint.yaml` file~~ Moved to #2403 
* [x] Check Keys and [annotations](https://squidfunk.github.io/mkdocs-material-insiders/reference/code-blocks/#inlinehilite) features
* [x] ~~split images into a root folder with common images and one for the specific versions for the rendered SVGs~~ Moved to #2403 
* [x] update footer according to the new landing page
    * [x] copyright
    * [x] vertical social media links
    * [x] improved logos
* [x] adapt script for new landing page -> Only `index.html` was changed, which is not updated by the script `tools/website/export-docu-to-website.sh` 

## Common

* [x] Update architecture diagrams
* [x] Revise `doc/website/getting-started/installation.md`
* [x] Revise `tools/website/how-to-record-asciicast.md`
* [x] Write down questions that came up before in `FAQ.md`
* [x] Decision whether to remove the QAC Helix guide (Can this be maintained by Perforce?) `tools/HelixQAC/README.md`
* [x]  Add integration hints to other platforms
* [x]  Add separator for the API change sections
* [x]  ~~icedelivery: Explain lifetime semantics of chunks/samples and user restrictions: loan needs to be followed with publish or release, MAX_CHUNKS_IN_USE on publisher and subscriber side etc., API is not thread-safe, update asciinema record (@MatthiasKillat)~~ Moved to #2403 
* [x] Add Apex.AI logo in `.svg`

## Examples

* [x] ~~Rework `icediscovery` example~~ Moved to #2403 
* [x] ~~Fix broken example `SingleProcess` on Windows~~ Fixed in #1082

## Update

* [x] Update `installation-guide-for-contributors.md` article

## New articles

* [x] Add binding/extension article about Rust, ara::com and DDS Gateway
  * [x] Update `iceoryx_components_diagram_nextgen_release.puml` e.g. removing DDS Gateway
* [x]  ~~How to write integration tests~~  Moved to #2403
* [x]  ~~Performance results section~~ Moved to #2403
* [x] Add an article which describes den command line parameters of RouDi
* [ ] Add new article describing the long-term goals of `iceoryx2` and the transition strategy from the user perspective

## Release note

* [x] Add entry about `clang-tidy` and `AUTOSAR` warning fixes
* [x] Cherry-pick `v2.x.y` release notes to main branch

## Last tasks

* [x] ~~Re-record all changed asciinema recordings~~ Not necessary for only non-functional RouDi changes
    * #971 ~~changed the output of all `iox-roudi` recordings, maybe record roudi startup only once to avoid to record all examples when the roudi output changes.~~

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] ~~Code follows the coding style of [CONTRIBUTING.md][contributing]~~
1. [x] ~~Tests follow the [best practice for testing][testing]~~
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

* Fix typos
* Fix line length
* Rename event id of 'Radar' service to 'Objects'

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1295  (partly)

### CODE REVIEW

File: iceoryx_examples/icediscovery/iox_find_service.cpp [hunk 1]
```diff
@@ -43,11 +43,11 @@ int main()
     {
         std::cout << "\n=========================================" << std::endl;
 
-        std::cout << "\nSearched for {'Radar', 'FrontLeft', 'Image'}. Found the following services:" << std::endl;
+        std::cout << "\nSearched for {'Radar', 'FrontLeft', 'Objects'}. Found the following services:" << std::endl;
         //! [search for unique service]
         serviceDiscovery.findService(iox::capro::IdString_t{"Radar"},
                                      iox::capro::IdString_t{"FrontLeft"},
-                                     iox::capro::IdString_t{"Image"},
+                                     iox::capro::IdString_t{"Objects"},
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/icediscovery/iox_find_service.cpp:50]: Please update the integration test

File: iceoryx_examples/icediscovery/README.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ It is included via:
 On that object we can call the method `findService` which expects the three
 service [string identifiers](https://github.com/eclipse-iceoryx/iceoryx/blob/v2.0.0/doc/website/getting-started/overview.md#creating-service-descriptions-for-topics)
 and a callable which will be applied to all matching services.
-In addition we have to specify whether we want to search for publishers (`MessagingPattern::PUB_SUB`)
+In addition, we have to specify whether we want to search for publishers (`MessagingPattern::PUB_SUB`)
```
Comment (MatthiasKillat) [inline iceoryx_examples/icediscovery/README.md:49]: Those commas are not needed (not wrong, but also not usually used unless we want to really emphasize what follows). Same for many others.

Whatever.

**Review by MatthiasKillat — CHANGES_REQUESTED** [review]
If we change the outputs (why?), we need to update the integration tests ...

**Review by MatthiasKillat — DISMISSED** [review]

**Comment by elBoberido** [comment]
Please change the target branch to master

**Comment by mossmaurice** [comment]
> Please change the target branch to master

@elBoberido I don't think that's a good idea. I propse to bring the certain small bug/doc fixes onto `release_2.0` first, and then backport them to `master`.

**Comment by mossmaurice** [comment]
As discussed, we won't do any major changes on the `release_2.0`. Hence, I changed the target branch to `master`.

File: iceoryx_examples/icediscovery/iox_find_service.cpp (hunk 1 above)
Comment (elfenpiff) [inline iceoryx_examples/icediscovery/iox_find_service.cpp:50]: With this change the C++ example is out of sync with the C example. Could you please make this change also in the c icediscovery example and adjust the explanation in the readme as well (also the integration tests for the c example).

But the easier approach would be to revert this change back to `Image` and you spare us reviewers and you some work.

**Review by elfenpiff — CHANGES_REQUESTED** [review]
When the `find

File: iceoryx_examples/icediscovery/iox_find_service.cpp (hunk 1 above)
Comment (mossmaurice) [reply iceoryx_examples/icediscovery/iox_find_service.cpp:50]: Thanks for the hint. I'll adapt corresponding c example and its integration tests as radar sensors typically don't send out images.

**Comment by mossmaurice** [comment]
I'll re-work this as part of #1295 and close this PR once there is a new one.

**Comment by elfenpiff** [comment]
> I'll re-work this as part of #1295 and close this PR once there is a new one. Please do not review.

@mossmaurice 

Wouldn't it be much better for the reviewers when you would use this PR to make the changes. Otherwise comments are lost and we may loose some hints and suggestions from this PR since they are not transferred.

**Comment by mossmaurice** [comment]
@elfenpiff 
> Wouldn't it be much better for the reviewers when you would use this PR to make the changes. Otherwise comments are lost and we may loose some hints and suggestions from this PR since they are not transferred.

I'm torn between here. Of course you are right, closing a PR and opening a new one is unhandy because developers have to keep track of findings in both PRs. On the other hand my intention for closing this PR here was to ensure proper traceability. The branch name still contains a issue #743 which was closed after the Blueberry release and should not be used for `v3.0`. I adapted all commit messages to the new release validation issue #1295 so it's fine IMHO traceability-wise.

Please go ahead with the review if you agree, otherwise I can also close and re-open with the correct branch name.

File: iceoryx_examples/icediscovery/README.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ It is included via:
 On that object we can call the method `findService` which expects the three
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/icediscovery/README.md:47]: What do you think about commenting out the expected output section since the asciicast is not in sync with your changes?

File: iceoryx_examples/icediscovery_in_c/README.md [hunk 4]
```diff
@@ -42,10 +42,10 @@ We can now call three different find service functions. Let's start with
 <!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][find service and apply callable]-->
 ```c
 iox_service_discovery_find_service_apply_callable(
-    serviceDiscovery, "Radar", "FrontLeft", "Image", printSearchResult, MessagingPattern_PUB_SUB);
+    serviceDiscovery, "Radar", "FrontLeft", "Objects", printSearchResult, MessagingPattern_PUB_SUB);
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/icediscovery_in_c/README.md:45]: same here

File: iceoryx_examples/icediscovery/README.md (hunk 3 above)
Comment (mossmaurice) [reply iceoryx_examples/icediscovery/README.md:47]: Good point, I'll add a todo to re-record it as we've done it before.

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

File: iceoryx_examples/icediscovery_in_c/README.md [hunk 5]
```diff
@@ -42,10 +46,10 @@ We can now call three different find service functions. Let's start with
...
```
Comment (MatthiasKillat) [inline iceoryx_examples/icediscovery_in_c/README.md:49]: I know we had this discussion but please note that here actually are radar images in the scientific literature. They are just tensors of F^nxmxk. With F some field, mainly real or complex numbers.

E.g. https://en.wikipedia.org/wiki/Synthetic-aperture_radar.

It was perfectly correct, and this change has no value. For the example it does not matter so, so we keep it now.

**Review by MatthiasKillat — APPROVED** [review]

File: iceoryx_examples/icediscovery_in_c/README.md (hunk 5 above)
Comment (mossmaurice) [reply iceoryx_examples/icediscovery_in_c/README.md:49]: @MatthiasKillat I know what you mean. IMHO what we demonstrate here is an example right before the perception module rather in the middle of the feedback loop and not in the front where the low-level computation on the raw radar data happens. Typically radar sensors will only provide object lists over the network to be consumed by central ADAS.
